subprojects/docs/docs.gradle

```diff
@@ -266,8 +267,12 @@ task dslHtml(type: Docbook2Xhtml) {
     ext.entryPoint = "$destDir/index.html"
 }
 
+task checkSectionIds(type: UserGuideSectionVerifierTask) {
+    docbookFiles = fileTree(userguideSrcDir) { include "**/*.xml" }
+}
+
 // This is used in the distribution and for the online version
-task userguideDocbook(type: UserGuideTransformTask, dependsOn: [samples, samplesDocbook, defaultImports]) {
+task userguideDocbook(type: UserGuideTransformTask, dependsOn: [checkSectionIds, samples, samplesDocbook, defaultImports]) {
```

Comment:
I'm not sure of the best way to do this. The only reliable way in my mind is to apply `mustRunAfter(checkSectionIds)` to all the other tasks that `userguideHtml` depends on.